common/client.go

```diff
@@ -326,12 +331,34 @@ func (c *DatabricksClient) niceAuthError(message string) error {
 	return fmt.Errorf("%s%s. Please check %s for details", message, info, docUrl)
 }
 
-func (c *DatabricksClient) fixHost() {
-	if c.Host != "" && !(strings.HasPrefix(c.Host, "https://") || strings.HasPrefix(c.Host, "http://")) {
-		// azurerm_databricks_workspace.*.workspace_url is giving URL without scheme
-		// so that is why this line is here
-		c.Host = "https://" + c.Host
+func (c *DatabricksClient) fixHost() error {
+	// Nothing to fix if the host isn't set.
+	if c.Host == "" {
+		return nil
+	}
+
+	u, err := url.Parse(c.Host)
+	if err != nil {
+		return err
+	}
+
+	// If the host is empty, assume the scheme wasn't included.
+	if u.Host == "" {
+		u, err = url.Parse("https://" + c.Host)
+		if err != nil {
+			return err
+		}
```

Comment:
this is strange, as invocation will always error out because of `url.Parse("https://")`

Reply:
No, u is the parsed URL, c is the input URL.

Reply:
ah, it's `u`, i've read it as `c`.